tests/test_join.py

```diff
@@ -0,0 +1,266 @@
+import pytest
+
+import greenplumpython as gp
+
+
+@pytest.fixture
+def db() -> gp.Database:
+    db = gp.database(host="localhost", dbname="gpadmin")
+    yield db
+    db.close()
+
+
+@pytest.fixture
+def t1(db: gp.Database):
+    # fmt: off
+    rows1 = [(1, 0, "'a1'",), (2, 0, "'a2'",), (3, 0, "'a3'",)]
+    # fmt: on
+    t = gp.values(rows1, db=db).save_as("temp1", temp=True, column_names=["id1", "idd1", "n1"])
+    return t
+
+
+@pytest.fixture
+def t2(db: gp.Database):
+    # fmt: off
+    rows2 = [(1, 0, "'b1'",), (2, 0, "'b2'",), (3, 0, "'b3'",)]
+    # fmt: on
+    t = gp.values(rows2, db=db).save_as("temp2", temp=True, column_names=["id2", "idd2", "n2"])
+    return t
+
+
+@pytest.fixture
+def zoo_1(db: gp.Database):
+    # fmt: off
+    rows = [(1, "'Lion'",), (2, "'Tiger'",), (3, "'Wolf'",), (4, "'Fox'")]
+    # fmt: on
+    t = gp.values(rows, db=db).save_as("zoo1", column_names=["id", "animal"])
+    return gp.table("zoo1", db)
+
+
+@pytest.fixture
+def zoo_2(db: gp.Database):
+    # fmt: off
+    rows = [(1, "'Tiger'",), (2, "'Lion'",), (3, "'Rhino'",), (4, "'Panther'")]
+    # fmt: on
+    t = gp.values(rows, db=db).save_as("zoo2", column_names=["id", "animal"])
+    return gp.table("zoo2", db)
+
+
+def test_join_default_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    ret = t1._join(
+        t2,
+        how="INNER JOIN",
+        on_str="ON temp1.id1 = temp2.id2",
+    ).fetch()
+    assert list(list(ret)[0].keys()) == ["id1", "idd1", "n1", "id2", "idd2", "n2"]
+
+
+def test_join_both_empty_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    ret = t1._join(t2, targets=[], how="INNER JOIN", on_str="ON temp1.id1 = temp2.id2").fetch()
+    assert list(list(ret)[0].keys()) == ["id1", "idd1", "n1", "id2", "idd2", "n2"]
+
+
+def test_join_all_emp_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
```

Comment:
`test_join_all_emp_targets` what does `emp` mean here? typo?

Reply:
> `test_join_all_emp_targets` what does `emp` mean here? typo?

One table with all columns and one table with empty targets list